src/LibraryManager.Vsix/Contracts/HostInteraction.cs

```diff
@@ -2,94 +2,121 @@
+
+            foreach (string filePath in relativeFilePaths)
             {
-                string absoluteFile = new FileInfo(Path.Combine(WorkingDirectory, relativeFilePath)).FullName;
+                string absoluteFilePath = Path.Combine(WorkingDirectory, filePath);
+                FileInfo file = new FileInfo(absoluteFilePath);
 
-                try
+                if (file.Exists)
                 {
-                    ProjectItem item = VsHelpers.DTE.Solution.FindProjectItem(absoluteFile);
-                    Project project = item?.ContainingProject;
-
-                    if (project != null)
-                    {
-                        item.Delete();
-                    }
-                    else
-                    {
-                        if (File.Exists(absoluteFile))
-                        {
-                            VsHelpers.CheckFileOutOfSourceControl(absoluteFile);
-                            File.Delete(absoluteFile);
-                        }
-                    }
-
-                    Logger.Log(string.Format(LibraryManager.Resources.Text.FileDeleted, relativeFilePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
-                }
-                catch (Exception ex)
-                {
-                    Logger.Log(string.Format(LibraryManager.Resources.Text.FileDeleteFail, relativeFilePath.Replace(Path.DirectorySeparatorChar, '/')), LogLevel.Operation);
-                    Telemetry.TrackException("deletefilefailed", ex);
+                    absolutePaths.Add(absoluteFilePath);
                 }
             }
+
+            //Delete from project 
+            var logAction = new Action<string, LogLevel>((message, level) => { Logger.Log(message, level); });
+            Project project = VsHelpers.GetDTEProjectFromConfig(_configFilePath);
+            bool deleteFromProject = await VsHelpers.DeleteFilesFromProjectAsync(project, absolutePaths, logAction, cancellationToken);
+            if (deleteFromProject)
+            {
+                return true;
+            }
+
+            // Delete from file system 
+
+            return await DeleteFilesFromDisk(absolutePaths, cancellationToken);
+
+        }
+
+        public Task<Stream> ReadFileAsync(string filePath, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            return FileHelpers.ReadFileAsStreamAsync(filePath, cancellationToken);
+        }
+
+        public Task<bool> CopyFile(string sourcePath, string destinationPath, CancellationToken cancellationToken)
+        {
+            return System.Threading.Tasks.Task.Run(() =>
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                return FileHelpers.CopyFile(sourcePath, destinationPath);
+            }, cancellationToken);
+        }
+
+        private Task<bool> DeleteFilesFromDisk(IEnumerable<string> filePaths, CancellationToken cancellationToken)
```

Comment:
Is this needed if it's just a wrapper?  Should `FileHelpers.DeleteFiles` be async instead?

Reply:
No, we can remove this method and call FileHe